client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/app/Game/GameBoard.java

```diff
@@ -0,0 +1,275 @@
+package edu.colostate.cs.cs414.chesshireCoders.jungleClient.app.Game;
+
+public class GameBoard {
+	//Set up Pieces
+	GamePiece[][] gamePieces = new GamePiece[2][8];
+	
+	//Set up Squares
+	BoardSquare[][] boardSquares = new BoardSquare[7][9];
+
+	public GameBoard() {
+		setUpBoard();
+		setUpPieces();
+	}
+	
+	public GameBoard(GamePiece[][] gamePieces) {
+		setUpBoard();
+		this.gamePieces = gamePieces;
+	}
+	
+	//the row and column here could be substituted for a direction, thought that would mean a minor loss of clarity.
+	public boolean movePiece(PieceType type, PlayerColor color, int row, int column) {
+		//WRITE TESTS then construct
+		//This method should check the movement with the checkMovement method
+		//Then apply the movement to the correct piece.
+		
+		/*
+		for(int i = 0; i < gamePieces.length; i++) {
+			if(gamePieces[i].getPieceType().equals(type)) {
+				if(gamePieces[i].getColor().equals(color)) {
+					if(!checkMovement(type, color, row, column)) {
+						return false;
+					}
+				}
+			}
+		}
+		*/
+		return false;
+	}
+	
+	//CheckMove will check that the move player's piece is allowed to move to the designated location.
+	//color must be 0 (red) or 1 (black)
+	//piece must be the powerlevel of the piece (aka Rat = 1 NOT 0)
+	//column = X axis location
+	//row = Y Axis Location
+	public boolean checkMove(int color, int piece, int column, int row) {		
+		//Assume piece given is in power level not array value
+		piece -= 1;		
+		if (piece < 0) {
+			return false;
+		}
+		
+		//finding the piece is dependent on what the UI/group wants. 
+		GamePiece gamePiece = gamePieces[color][piece];
+		int pieceColumn = gamePiece.getColumn();
+		int pieceRow = gamePiece.getRow();
+		
+		//Check the piece is not dead
+		if(gamePieces[color][piece] == null){
+			return false;
+		}
+		
+		//Only the row or column can change, one must change.
+		if(((column == pieceColumn) && (row == pieceRow))||((column != pieceColumn) && (row != pieceRow))){
+			return false;
+		}
+		
+		//The only time the row or column will increase by more than one is if the tiger or leopard is jumping the river
+		//Check if it increases by more than one
+		if(!(((pieceColumn+1 == column)||(pieceColumn-1 == column))||((pieceRow+1 == row)||(pieceRow-1 == row)))){
+			//if it is not a tiger or a leopard return false
+			if(!((piece == 4)||(piece == 5))) {
```

Comment:
I can see the argument here, however as I can see the code will remain the same, it can be moved if need be, but it's the same either way, and this way reduces coupling while keeping the board design in once place.